python/paddle/fluid/tests/unittests/test_activation_op.py

```diff
@@ -895,6 +896,58 @@ def test_check_grad(self):
         self.check_grad(['X'], 'Out')
 
 
+class TestTan(TestActivation):
+    def setUp(self):
+        np.random.seed(1024)
+        self.op_type = "tan"
+        self.init_dtype()
+        self.dtype = 'float32'
+        self.x_np = np.random.uniform(-1, 1, [10, 12]).astype(self.dtype)
+        self.place = paddle.CUDAPlace(0) if core.is_compiled_with_cuda() \
+            else paddle.CPUPlace()
+
+        out = np.tan(self.x_np)
+
+        self.inputs = {'X': OpTest.np_dtype_to_fluid_dtype(self.x_np)}
+        self.outputs = {'Out': out}
+
+    def test_check_grad(self):
+        if self.dtype == np.float16:
+            return
+        self.check_grad(['X'], 'Out')
+        print(self.check_grad(['X'], 'Out'))
+
+    def test_dygraph_api(self):
+        paddle.disable_static(self.place)
+        x = paddle.to_tensor(self.x_np)
+        out_test = paddle.tan(x)
+        out_ref = np.tan(self.x_np)
+        self.assertEqual(np.allclose(out_ref, out_test.numpy()), True)
+        paddle.enable_static()
+
+    def test_static_api(self):
+        paddle.enable_static()
+        with paddle.static.program_guard(paddle.static.Program()):
+            x = paddle.static.data('X', [10, 12], self.dtype)
+            out = paddle.tan(x)
+            exe = paddle.static.Executor(self.place)
+            res = exe.run(feed={'X': self.x_np}, fetch_list=[out])
+        out_ref = np.tan(self.x_np)
+        self.assertEqual(np.allclose(out_ref, res[0]), True)
```

Comment:
Not important, you can change or not.

You can write self.assertTrue(np.allclose(out_ref, out_test.numpy()))

Reply:
thx, done